### DISCUSSION TITLE
Create new column in taskwarrior with total time spent of the project ( tag )

### DISCUSSION THREAD

**Discussion by gkssjovi** [GothenburgBitFactory/taskwarrior#2657] [Q&A] [answered]
Hello,

How I can create a new column in taskwarrior that will output the total time spent on the project  from timewarrior?
Basically the column output should be the result of executing this code ` timew summary Inbox`

**Comment by smemsh** [comment, marked as answer]
This is impossible, especially since there's no 1:1 equivalence between tasks and timew intervals.  But also because there's no way to "import" data from elsewhere into a task report column.  You would need to write your own report that tied together timew export and taskw export, I suppose?

**Reply by gkssjovi** [reply]
I have made this possible for anyone how want this feature. I have put the `taskwarrior` uuid as a tag in `timewarrior`, in this way I was able to track the time related to the task.

https://github.com/gkssjovi/trackwarrior

**Reply by smemsh** [reply]
Yes, I do a similar thing, I have a UDA that I maintain uniquely in taskwarrior and I add a tag with that in timewarrior.  Actually the timewarrior tag is a concatenation of project hierarchy separated by slashes, along with the UDA.  Probably you could use the first 32 bits of uuid and this would be good and short enough also.

I did not think to insert timew data back into taskwarrior in timew extension,  That's a good idea.  But then you have to run a timew report to get the data into taskwarrior, yes? Since timewarrior does not have hooks per se, only custom reports.

**Reply by gkssjovi** [reply]
The idea is to see the total time spent, and the cost in taskwarrior without using the timewarrior.
Using the short UUID is good enough, without any concatenation because in this way you can modify the description of the task add or remove tags, and the time spent will not change. Also If I want to clear the time, I can add a tag +cleartime and the time will be set to 0.

**Reply by smemsh** [reply]
> The idea is to see the total time spent, and the cost in taskwarrior without using the timewarrior

You still use timewarrior to start/stop/continue though right? So the data can only be put into taskwarrior via a timewarrior extension.  These are only invoked as separate commands.  There is no way to hook start/stop/continue for example so the insertion into taskwarrior can be done automatically.

**Reply by gkssjovi** [reply]
I don't use timewarrior. Only taskwarrior. The hooks in taskwarrior will automatically  start the timewarrior, and the total time from timewarrior will be visible on the task.

**Comment by gkssjovi** [comment]
I think you ware right, this will require some work to achieve that.
thx.
